sips/sip-022/sip-022-emergency-pox-fix.md

```diff
@@ -0,0 +1,235 @@
+  block before the activation height.  Given that the set of solo stackers in
+  cycle #56 is small and fixed, the additional overhead of fixing all records in
+  `reward-cycle-pox-address-list` will be minimal.  Nevertheless, as a safety
+  precaution, each Stacks block whose parent was mined prior to the Bitcoin
+  block activation height must contain zero transactions and confirm zero
+  microblocks.
+
+* Set the minimum required block-commit memo bits to `0x07`.  All block-commits
+  after the Bitcoin block activation height must have a memo value of at least
+`0x07`.  This ensures that miners that do not upgrade from Stacks 2.1 will not
+be able to mine in Stacks 2.2.
+
+* Set the peer network version bits to `0x18000007`.  This ensures that follower
+  nodes that do not upgrade to Stacks 2.2 will not be able to talk to Stacks
+2.2 nodes.
+
+# Related Work
+
+Consensus bugs requiring immediate attention such as this
+have been detected and fixed in other blockchains.  In the
+absence of a means of gathering user comments on proposed fixes, the task of
+activating these bugfixes has fallen to miners, exchanges, and node runners.  As
+long as sufficiently many participating entities upgrade, then a chain split is
+avoided and the fixed blockchain survives.  A prominent example was Bitcoin
+[CVE-2010-5139](https://www.cvedetails.com/cve/CVE-2010-5139/), in which a
+specially-crafted Bitcoin transaction could mint arbitrarily many BTC well above
+the 21 million cap.  The [developer
+response](https://bitcointalk.org/index.php?topic=823.0) was to quickly release
+a patched version of Bitcoin and rally enough miners and users to upgrade.  In a
+matter of hours, the canonical Bitcoin chain ceased to include any transactions
+that minted too much BTC.
+
+# Backwards Compatibility
+
+The activation of this SIP does _not_ mean that the ability to increase one's locked STX is 
+disabled.  The `delegate-stack-increase` function is not affected by this bug.
+Solo stackers who want to contineu increasing their locked STX will need to
+delegate their STX to an address they control, and use this API to increase
+their locked STX going forward.
+
+There are no changes to the chainstate database schemas in this SIP.  Everyone
+who runs a Stacks 2.1 node today will be able to run a Stacks 2.2 node off of
+their existing chainstates.
+
+Stacks 2.2 nodes will not interact with Stacks 2.1 nodes on the peer network
+after the Bitcoin block activation height passes.  In addition, Stacks 2.2 nodes
+will ignore block-commits from Stacks 2.1 nodes.  Similar changes were made for
+Stacks 2.05 and Stacks 2.1 to ensure that the new network cleanly separates from
+stragglers still following the old rules.
+
+# Activation
+
+The Bitcoin block activation height will need to pass prior to the selection of
+the PoX anchor block for reward cycle #57 (Bitcoin block height 785751).  This
+SIP proposes Bitcoin block height 785551, which is 200 Bitcoin blocks prior.  In
```

Comment:
I think this should be reward cycle 58 and bitcoin blocks 787751 and 787551, respectively.

Reply:
Earlier there's a reference to 785,550 too.

Reply:
Yup